Yoast/javascript
# P1-323 Fix keyboard trap in replacement variable editors

## Summary


This PR can be summarized in the following changelog entry:

* [@yoast/replacement-variable-editor] Fixes a bug where the cursor could become trapped in some of our input fields when using keyboard navigation.
* [yoast-components] NON-USER-FACING Removed unused Draft-JS and plugin dependencies.

## Relevant technical choices:

* Our draft-js-plugins dependency has been outdated for a while, and I wanted to try whether updating it would fix the keyboard lock. The update was a "major" one, so it required some additional changes.
* Somehow tabbing places the cursor at a field's beginning now, I haven't been able to fix this, but it seems less annoying than a keyboard trap. Progress?
* Removed unused draft-js and plugins from `yoast-components`.

## Test instructions

This PR can be tested by following these steps:

* Checkout this branch and the companion branch in free.
* Link monorepo.
* Go to a post editor, and mouse click in the Focus keyphrase field in the metabox.
* Now start tabbing to the snippet editor fields (seo title and description).
* Verify that you can navigate in out of those fields by tabbing and shift+tabbing.
* Verify that the replacement variables and the input fields work and look as before in all other regards.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * Every replacement variable editor:
  	* Snippet Editor
  	* Social Preview forms
  	* Replacement variable editors in Yoast settings.

## UI changes

## Quality assurance


Fixes P1-179
